DIPs/DIP1xxx-rn.md

```diff
@@ -0,0 +1,478 @@
+    this.tupleof[i] = field;
+```
+
+As shown above, the single benefit of the postblit can be easily substituted with a
+few lines of code. On the other hand, to fix the limitations of the postblit it is
+required that more complexity is added for little to no benefit. In these circumstances,
+for the sake of uniformity and consistency, replacing the postblit constructor with a
+copy constructor is the reasonable thing to do.
+
+## Description
+
+This section discusses all the technical aspects regarding the semantics
+of the copy constructor.
+
+### Syntax
+
+The following link exhibits the proposed syntax and the necessary grammar changes in order to support
+the copy construct:
+
+https://github.com/dlang/dlang.org/pull/2414
+
+The copy constructor declaration will be recognized by the parser when the tokens `@`, `implicit`, `this`
+are encountered exactly in this order. The use of a specific user defined attribute in conjunction with `this`
+benefits from the advantage of declaring the copy constructor in an expressive manner without adding any new
+keywords to the language.
+
+The argument to the copy constructor is passed by reference in order to avoid infinite recursion (passing by
+value would require a copy of the `struct` which would be made by calling the copy constructor, thus leading
+to an infinite chain of calls).
+
+The type qualifiers may be applied to the parameter of the copy constructor, but also to the function itself
+in order to facilitate the ability to describe copies between objects of different mutability levels. The type
+qualifiers are optional.
+
+### Semantics
+
+This sections discusses all the aspects regarding the semantic analysis and interaction with other
+components of the copy constructor
+
+#### Requirements
+
+1. The `structName` type needs to be identical to `typeof(this)`; an error is issued otherwise. This
+requirement may be relaxed in the future in order to accomodate implicit copying from objects of a different type:
+
+```d
+struct A
+{
+    int a;
+    string b;
+}
+
+struct B
+{
+    A a;
+    int c;
+}
+
+struct C
+{
+    A a;
+    string b;
+    @implicit this(ref C another)       // ok, typeof(this) == C
+    {
+        this.a = another.a;
+        this.b = another.b;
+    }
+
+    @implcit this(ref A another)       // error typeof(this) != C, but might be accepted in the future
+    {
+        this.a = another;
+        this.b = "hello";
+    }
+}
+
+void main()
+{
+    C a, b;
+    A aa;
+
+    a = b;        // ok
+    a = aa;       // error
+}
+```
+2. It is illegal to declare a copy constructor for a struct that has a postblit defined and vice versa:
+
+```d
+struct A
+{
+    this(this) {}
+    @implicit this(ref A another) {}        // error, struct A defines a postblit
+}
+```
+
+#### Semantics
+
+The copy constructor typecheck is identical to the constructor one [6][7]. If a copy constructor
+has an empty body, the source is initialized to its `.init` property.
```

Comment:
There is no order, because the user is in charge of copying all the fields of `B`. If inside the copy constructor there is a line "this.aaa = rhs.aaa", then A's copy constructor will get called (if A defines a copy-ctor)